### PR TITLE
Refactor ClinicServiceImpl

### PR DESCRIPTION
**Pull request by earlspilner**
### Summary

This PR makes all fields in the `ClinicServiceImpl` class `final` to ensure they are immutable after initialization. It also removes some redundant imports.

### Changes

- Made fields in `ClinicServiceImpl` `final`.
- Updated the constructor to initialize these fields.
- Removed unnecessary imports.

- UPD: added private helper method to extract common entity retrieval logic

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java [hunk 1]
```diff
@@ -49,248 +39,248 @@
...
-    private SpecialtyRepository specialtyRepository;
-	private PetTypeRepository petTypeRepository;
+    private final PetRepository petRepository;
+    private final VetRepository vetRepository;
+    private final OwnerRepository ownerRepository;
+    private final VisitRepository visitRepository;
+    private final SpecialtyRepository specialtyRepository;
+    private final PetTypeRepository petTypeRepository;
 
     @Autowired
```
Comment (arey) [inline src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java:49]: nitpick: by the way, we could remove this uncessary `@Autowired`

File: src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java [hunk 2]
```diff
@@ -49,248 +39,248 @@
...
+
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Collection<Owner> findOwnerByLastName(String lastName) throws DataAccessException {
+        return ownerRepository.findByLastName(lastName);
     }
 
-	@Override
```
Comment (arey) [inline src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java:75]: thought: it seems, there are a lot of reformatting for converting tab to spaces according to the `.editorconfig`

File: src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java (hunk 1 above)
Comment (earlspilner) [reply src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java:49]: I agree that this annotation is optional, but for readability, as well as to make it easier for a beginner to figure out how it is injected, it seems to me better to leave this annotation as it is.

**Comment by earlspilner** [comment]
I've added commit for [Issue #144](https://github.com/spring-petclinic/spring-petclinic-rest/issues/144) there, @arey

File: src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java (hunk 1 above)
Comment (arey) [reply src/main/java/org/springframework/samples/petclinic/service/ClinicServiceImpl.java:49]: I don't agree, but that's debatable.
For your information, the canonical version of Spring Petclinic didn't use the `@Autowired`
https://github.com/spring-projects/spring-petclinic/blob/main/src/main/java/org/springframework/samples/petclinic/owner/OwnerController.java

**Comment by arey** [comment]
Thanks a lot Alexander :D

**Pull request merged**